### PR TITLE
Migrate to new 32blit wave API

### PR DESCRIPTION
**Pull request by Gadgetoid**
Update for the API changes in https://github.com/pimoroni/32blit-beta/pull/562

Doesn't.... seem to work... :grimacing:

### CODE REVIEW

File: src/blit/sound.cpp [hunk 1]
```diff
@@ -131,20 +131,20 @@ static void RefillBuffer(void *data)
...
-            blit::channels[channel].wave_buffer[i + 1] = sound.data[sound.offset] - 127;
+            channel.wave_buffer[i] = sound.data[sound.offset] - 127;
+            channel.wave_buffer[i + 1] = sound.data[sound.offset] - 127;
         }
     }
     else
     {
         for(i = 0; i < 64 && sound.offset < sound.length; i++, sound.offset++)
-            blit::channels[channel].wave_buffer[i] = sound.data[sound.offset] - 127;
+            channel.wave_buffer[i] = sound.data[sound.offset] - 127;
```
Comment (Daft-Freak) [inline src/blit/sound.cpp]: These will need a * 256 (I think, looks like 8-bit samples)

File: src/blit/sound.cpp [hunk 2]
```diff
@@ -155,8 +155,8 @@ static boolean I_Blit_InitSound(boolean _use_sfx_prefix)
     for(int i = 0; i < SOUND_CHANNELS; i++)
     {
         blit::channels[i].waveforms = blit::Waveform::WAVE;
-        blit::channels[i].wave_callback_arg = (void *)(uintptr_t)i;
-        blit::channels[i].callback_waveBufferRefresh = &RefillBuffer;
+        blit::channels[i].user_data = (void *)(uintptr_t)i;
```
Comment (Daft-Freak) [inline src/blit/sound.cpp]: Hmm, this could be `&channel_sounds[i]` now...

File: src/chocdoom/opl/opl_blit.cpp [hunk 3]
```diff
@@ -233,7 +233,7 @@ static int opl_blit_Init(unsigned int port_base)
 
     blit::channels[7].waveforms = blit::Waveform::WAVE;
     blit::channels[7].volume = 0xFF;
```
Comment (Daft-Freak) [inline src/chocdoom/opl/opl_blit.cpp]: Can drop setting the volume here now.

**Review by Daft-Freak — COMMENTED** [review]
Haha, I was going to fix all this... but ended up trying to play Super Mario Land on a Pico. :smile: 

Anyway, I think it's probably a volume problem.

**Comment by Gadgetoid** [comment]
Sound is now ... sounding... but even with Music Volume cranked up to max, it seems to be noticeably quieter than the SFX at 50%.

**Comment by Daft-Freak** [comment]
Sounds about the same as before... (Probably the magic number [here](https://github.com/Daft-Freak/32BlitDoom/blob/938ed701157312c6194d8899e5f8d42b7c7e2d1d/src/blit/sound.cpp#L185) needs adjusting, I just went for max loudness).

Thanks!

**Pull request merged**